### PR TITLE
feat(HNG-6): add a sign up otp modal

### LINKED ISSUE #6: [FEAT] - Authentication Page -> Sign up/OTP modal
### Description

Implement the sign up OTP Modal as shown in the design

### Acceptance Criteria

- Responsive across all screen sizes (desktop, tablet and mobile)
- Write tests for your code.
- Use components where necessary (header, footer, buttons, modal and so on)

### Testing

- Ensure the page renders correctly with all required elements.
- Validate the responsiveness and accessibility features.
- Check that the input handles the correct types.

### Tasks

- [ ] Develop the page based on the provided requirements.
- [ ] Implementation should be responsive across all screen sizes.
- [ ] Write unit tests to ensure the component works as expected.
- [ ] Validate the input field

[Link to Design](
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1-18104&t=p2gECcZQmYqvvrn7-4)

### Preview
image

### PR DESCRIPTION
**Pull request by Boblee24**
# Description

**Closes #6 **

# Changes proposed

## created a single component, work on the otp input and validation of the input so that it takes on number, can also paste number without issues

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
[scrnli_7_20_2024_10-19-29 PM.webm](https://github.com/user-attachments/assets/b16904b6-b66c-4bd9-8f4d-4ce48d9f67fc)

Screenshot (53)

### CODE REVIEW

**Comment by oreoluwa212** [comment]
Lint test failed

File: src/components/modals/signupmodal/SignUpOtp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { fireEvent, render, screen } from "@testing-library/react";
```
Comment (oreoluwa212) [inline src/components/modals/signupmodal/SignUpOtp.test.tsx]: Dont import anything from vitest, take it out. It has been done globally.

File: src/components/modals/signupmodal/signUpOtp.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    const pasteData = event.clipboardData.getData("text");
+    if (/^\d{6}$/.test(pasteData)) {
+      const newOtp = [...pasteData];
+      setOtp(newOtp);
+      for (const [index, value] of newOtp.entries()) {
+        if (inputReferences.current[index]) {
+          inputReferences.current[index]!.value = value;
+        }
+      }
+      inputReferences.current[5]?.focus(); // Focus the last input after pasting
```
Comment (oreoluwa212) [inline src/components/modals/signupmodal/signUpOtp.tsx:51]: Use shadcn for your modal and you'll be good to go

**Comment by Boblee24** [comment]
Passing lint on my machine but failing here, omooooo

**Pull request closed without merging**